ipn/ipnlocal/profiles.go

```diff
@@ -231,37 +233,69 @@ func (pm *profileManager) SetPrefs(prefsIn ipn.PrefsView, np ipn.NetworkProfile)
 				// We couldn't delete the state, so keep the profile around.
 				continue
 			}
-			// Remove the profile, knownProfiles will be persisted below.
+			// Remove the profile, knownProfiles will be persisted
+			// in [profileManager.setProfilePrefs] below.
 			delete(pm.knownProfiles, p.ID)
 		}
-	} else if cp.ID == "" {
-		// We didn't have an existing profile, so create a new one.
-		cp.ID, cp.Key = newUnusedID(pm.knownProfiles)
-		cp.LocalUserID = pm.currentUserID
-	} else {
-		// This means that there was a force-reauth as a new node that
-		// we haven't seen before.
```

Comment:
we lost this comment, where is this handled now?

Reply:
We didn't handle this scenario in any special way; it was an empty block.
I think its original purpose was to document when the two preceding conditions wouldn't be met, but I believe it's no longer relevant after the refactoring.

If we ever need to actually handle this scenario, we could check it in `SetProfilePrefs` like this:
```go
if persist := prefsIn.Persist(); profile.ControlURL != prefsIn.ControlURL() || (persist.Valid() && profile.NodeID != persist.NodeID()) {
    // This means that there was a force-reauth as a new node that
    // we haven't seen before.
}
```